src/main/java/com/muzima/view/forms/HTMLFormWebViewActivity.java

```diff
@@ -385,6 +385,7 @@ private FormData createNewFormData() throws FormController.FormDataSaveException
             setStatus(STATUS_INCOMPLETE);
             setTemplateUuid(form.getUuid());
```

Comment:
We can use this template uuid and send this as top level element in the FormData and process this in the server side. This would avoid the need to set and send form name as separate field.